File: leetcode/neetcode/arrays-hashing/1.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        
        NumsMap = {}

        for i, n in enumerate(nums):
            diff = target - n
            if diff in NumsMap:
                return [NumsMap[diff], i]
            NumsMap[n] = i

File: leetcode/neetcode/arrays-hashing/1.cpp
// 1. Two Sum

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> numsMap;
        vector<int> res; 

        for (int i = 0; i < nums.size(); i++) {
            int complem = target - nums[i];
            if (numsMap.find(complem) != numsMap.end()) {
                res.push_back(numsMap[complem]); 
                res.push_back(i); 
                return res; 
            }
            numsMap[nums[i]] = i; 
        }
        return res;
    }
};
